[Go/Python][Flaky Test] New DNS test broke test container tests

### System Information

test-go-container

### Language and Version

Go

### Engine Version

n/a

### Logs

```
Prepare all required actions
Run ./.github/actions/setup-dns-tests
  with:
    os: ubuntu
  env:
    CARGO_TERM_COLOR: always
    BASE_GO_VERSION: 1.22
    ENGINE_VERSION: 9.0
    CARGO_ZIGBUILD_TARGET: x86_64-unknown-linux-musl
Run ENTRIES="127.0.0.1 valkey.glide.test.tls.com
  ENTRIES="127.0.0.1 valkey.glide.test.tls.com
  127.0.0.1 valkey.glide.test.no_tls.com
  ::1 valkey.glide.test.tls.com
  ::1 valkey.glide.test.no_tls.com"
  if [[ "$OS" == "windows" ]]; then
      echo "$ENTRIES" >> /c/Windows/System32/drivers/etc/hosts
  else
      echo "$ENTRIES" | sudo tee -a /etc/hosts
  fi
  shell: bash --noprofile --norc -e -o pipefail {0}
  env:
    CARGO_TERM_COLOR: always
    BASE_GO_VERSION: 1.22
    ENGINE_VERSION: 9.0
    CARGO_ZIGBUILD_TARGET: x86_64-unknown-linux-musl
    OS: ubuntu
/__w/_temp/7d08a904-edf1-48d9-81de-fe11336f4b13.sh: line 8: sudo: command not found
Error: Process completed with exit code 127.
```

### Glide Version

main


### Actual Behavior

The setup-dns-tests composite action unconditionally uses sudo for non-Windows OS targets. This fails on container-based jobs (e.g., amazon-linux) where sudo is not installed and the process already runs as root.

### Possible Fixes

Check for sudo availability before using it, falling back to plain tee when running as root in a container

--- a/.github/actions/setup-dns-tests/action.yml
+++ b/.github/actions/setup-dns-tests/action.yml
@@ -26,6 +26,8 @@ runs:
               ::1 valkey.glide.test.no_tls.com"
               if [[ "$OS" == "windows" ]]; then
                   echo "$ENTRIES" >> /c/Windows/System32/drivers/etc/hosts
-              else
+              elif command -v sudo &> /dev/null; then
                   echo "$ENTRIES" | sudo tee -a /etc/hosts
+              else
+                  echo "$ENTRIES" | tee -a /etc/hosts
               fi

